framework/src/csifw/CSILogsDumper.c

```diff
@@ -0,0 +1,225 @@
+/****************************************************************************
+ *
+ * Copyright 2024 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+#include "include/CSILogsDumper.h"
+#include <sys/socket.h>
+#include <netdb.h>
+#include <pthread.h>
+#include <errno.h>
+
+#define PORT 5000
+#define CONTROL_COMMAND_BUFFER_SIZE 128
+#define CSI_DATA_DUMP_BUFFER_SIZE 1024
+
+unsigned char gCSIDataDumpBuffer[CSI_DATA_DUMP_BUFFER_SIZE];
+char gControlCommandBuffer[CONTROL_COMMAND_BUFFER_SIZE];
+
+csiDataDumpListener gListener = NULL;
+static pthread_t gThreadId = -1;
+static int gSockFd = -1, gFifoFd = -1;
+struct sockaddr_in gServerAddr, gClientAddr;
+socklen_t gClientLen = sizeof(gClientAddr);
+fd_set readfds;
+int maxfd;
+
+static int gWaitingForData = 0;
+static int gRemainingBytes = 0;
+static int gBytesReadSofar = 0;
+
+static int recv_message(int fd, void *buf, int buflen)
+{
+	int received = 0;
+	while (1) {
+		int res = read(fd, buf + received, buflen - received);
+		if (res < 0) {
+			int err_no = errno;
+			if (err_no == EAGAIN || err_no == EINTR) {
+				continue;
+			}
+			CSIFW_LOGE("read error %d\n", err_no);
+			return -1;
+		}
+		received += res;
+		if (received == buflen) {
+			break;
+		}
+	}
+	return received;
+}
+
+static void print_buf(const unsigned char* buf, int len)
+{
+	printf("\n RAW DATA %d\n\n", len);
+	unsigned long long *buff_tmp = (u64 *)buf;
+	int buff_len = (len / 8) + 1;
+	for (int i = 0; i < buff_len; i++) 

```

Comment:
according to our coding rule, add bracket even if it is single line of code.

Reply:
It is fixed